src/main/java/com/example/gateway/controllers/ItemController.java

```diff
@@ -0,0 +1,70 @@
+package com.example.gateway.controllers;
+
+import com.example.gateway.clients.ItemClient;
+import com.example.gateway.clients.UserClient;
+import com.example.gateway.exceptions.AccessDeniedException;
+import com.example.gateway.services.UserService;
+import com.example.lib.dto.create.ItemCreationRequest;
+import com.example.lib.dto.create.ItemCreationResponse;
+import com.example.lib.dto.read.ItemReadAllResponse;
+import com.example.lib.dto.update.ItemUpdateRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.UUID;
+
+import static com.example.gateway.constants.Constants.*;
+
+@RestController
+@RequestMapping("/items")
+@RequiredArgsConstructor
+public class ItemController {
+
+    private final ItemClient itemClient;
+    private final UserClient userClient;
+    private final UserService userService;
+
+    @GetMapping
+    public List<ItemReadAllResponse> findAll(@RequestParam(name = "page", defaultValue = "0", required = false) int page,
```

Comment:
Можно мапу тут использовать.
Тогда все параметры в мапе будут и не так объемно выглядеть будет.
Плюс не придется метод контроллера переписывать при добавлении новых параметров

Reply:
Resolved, but I decided to use another way to avoid this big part of code. I create like this:
`@GetMapping("/items")
    List<ItemReadAllResponse> findAll(@RequestParam(value = "page", required = false) Integer page,
                                      @RequestParam(value = "size", required = false) Integer size,
                                      @RequestParam(value = "sort", required = false) String sort,
                                      @SpringQueryMap ItemFilter filter);`
  in feign client, and this:
   `@GetMapping
    public ResponseEntity<List<ItemReadAllResponse>> findAll(
                                                             @RequestParam(value = "page", required = false) Integer page,
                                                             @RequestParam(value = "size", required = false) Integer size,
                                                             @RequestParam(value = "sort", required = false) String sort,
                                                             ItemFilter filter) {
        return new ResponseEntity<>(itemClient.findAll(page, size, sort, filter), HttpStatus.OK);
    }`
    in controller on gateway.

Reply:
Use Pagable instead  @RequestParam(value = "page", required = false) Integer page, @RequestParam(value = "size", required = false) Integer size, @RequestParam(value = "sort", required = false) String sort,